### PR TITLE
Feature/bigintbool

### PR DESCRIPTION
**Pull request by noambergIL**
Add support for bigint and bool + some test fixes for the input/output arguments.

### CODE REVIEW

File: services/processor/native/call.go [hunk 1]
```diff
@@ -174,15 +217,27 @@ func createMethodOutputArgs(methodInstance types.MethodInstance, args []reflect.
...
+				} else {
+					return nil, errors.Errorf("method '%s' output arg %d is of unsupported type %s", functionNameForErrors, i, arg.Type())
 				}
+			} else {
+				return nil, errors.Errorf("method '%s' output arg %d is of unsupported type %s", functionNameForErrors, i, arg.Type())
+			}
+		case reflect.Bool:
+			res = append(res, &protocol.ArgumentBuilder{Type: protocol.ARGUMENT_TYPE_BOOL_VALUE, BoolValue: arg.Interface().(bool)})
+		case reflect.Ptr:
+			if arg.Type() == bigIntType {
```
Comment (electricmonk) [inline services/processor/native/call.go]: maybe extract a function `isBigInt(arg)`?

File: services/processor/native/call_test.go [hunk 2]
```diff
@@ -1,174 +1,192 @@
...
 	"github.com/stretchr/testify/require"
+	"math/big"
 	"reflect"
 	"testing"
 )
 
-func TestPrepareMethodArgumentsForCallWithUint32(t *testing.T) {
-	methodInstance := func(a uint32) uint32 {
-		return a
+func TestPrepareMethodArgumentsAndCall_SimpleOneInputArg(t *testing.T) {
```
Comment (electricmonk) [inline services/processor/native/call_test.go]: I assume these new tests cover all the edge cases you handled?

File: services/processor/native/call.go [hunk 3]
```diff
@@ -93,9 +99,28 @@ func prepareMethodInputArgsForCall(methodInstance types.MethodInstance, args *pr
...
+			if methodTypeIn == bigIntType {
+				if !arg.IsTypeUint256Value() {
+					return nil, errors.Errorf("method '%s' expects arg %d to be *big.Int but it has %s", functionNameForErrors, i, arg.StringType())
 				}
+				res = append(res, reflect.ValueOf(arg.Uint256Value()))
+			} else {
+				return nil, errors.Errorf("method '%s' expects arg %d to be a known type but it has %s", functionNameForErrors, i, arg.StringType())
 			}
 		case reflect.Slice:
+			// Note we don't support []bool
```
[the remark this answers is not in the document]
Comment (noambergIL) [reply services/processor/native/call.go]: yes, but this is not even related. un-needed comment.

File: services/processor/native/call_test.go (hunk 2 above)
Comment (noambergIL) [reply services/processor/native/call_test.go]: the two table tests check only the simple version (one in/one out) as a way to show which types are supported. the more complex tests are for edge cases i could think of.
